### PR TITLE
ci: improve setup

### PR DESCRIPTION
**Pull request by AlexandruLazarescu-TomTom**
Ensure we are using Gemfile(.lock) for tracking
gem dependencies.
Lock pods spec checksums to improve package
security and integrity.
Use bundler as a driver for pod commands.
Improve GitHub workflow setup.

### CODE REVIEW

File: Gemfile.lock [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+      CFPropertyList (>= 2.3.3, < 4.0)
+      atomos (~> 0.1.3)
+      claide (>= 1.0.2, < 2.0)
+      colored2 (~> 3.1)
+      nanaimo (~> 0.3.0)
+      rexml (~> 3.2.4)
+    zeitwerk (2.6.8)
+
+PLATFORMS
+  x86_64-darwin-21
```
Comment (muhammedtanriverdi-tomtom) [inline Gemfile.lock:92]: Is it possible to add other platforms? 
- arm64-darwin-22

**Review by muhammedtanriverdi-tomtom — APPROVED** [review]
It looks much better than before 👍 
I have a few minor comments that I would like to share.

File: README.md [hunk 2]
```diff
@@ -75,12 +101,23 @@ Warning: Error creating LLDB target at path '*' using an empty LLDB target which
 
 ### Apple M1 and M2 processor support
```
[the remark this answers is not in the document]
Comment (AlexandruLazarescu-TomTom) [reply README.md]: done in d8d56d2500439a0e2ff0f8fe37012903d1d0a2e2

File: README.md [hunk 3]
```diff
@@ -75,12 +101,23 @@ Warning: Error creating LLDB target at path '*' using an empty LLDB target which
...
 
-Xcode requires a specific setup to support TomTom SDK on Apple M1 and M2 processors devices. The existing frameworks are optimized for Intel processors. Apple introduced Rosetta to emulate devices with Apple M1 and M2 processors, using Intel processors for backward compatibility. Please follow the steps below to run Rosetta during the build process:
+Xcode requires a specific setup to support TomTom SDK on Apple M1 and M2 processors devices.
```
[the remark this answers is not in the document]
Comment (AlexandruLazarescu-TomTom) [reply README.md]: done in d8d56d2500439a0e2ff0f8fe37012903d1d0a2e2

File: README.md [hunk 4]
```diff
@@ -75,12 +101,23 @@ Warning: Error creating LLDB target at path '*' using an empty LLDB target which
...
+The existing frameworks are optimized for Intel processors. Apple introduced Rosetta to emulate devices with
+Apple M1 and M2 processors, using Intel processors for backward compatibility.
```
[the remark this answers is not in the document]
Comment (AlexandruLazarescu-TomTom) [reply README.md]: done in done in d8d56d2500439a0e2ff0f8fe37012903d1d0a2e2

File: Gemfile.lock (hunk 1 above)
Comment (AlexandruLazarescu-TomTom) [reply Gemfile.lock:93]: done in d8d56d2500439a0e2ff0f8fe37012903d1d0a2e2

File: README.md [hunk 5]
```diff
@@ -1,59 +1,85 @@
...
+   4. Load `rbenv` to your shell
+   ```zsh
+   echo 'eval "$(rbenv init - zsh)"' >> ~/.zshrc
+   ```
+   Above command assumes that `zsh` is being used. If a different shell is being used, then you can find 
+   commands for most common shells in [rbenv documentation].
+
+   5. Update `bundler`
+   ```zsh
+   gem install bundler -v 2.3.23
```
[the remark this answers is not in the document]
Comment (AlexandruLazarescu-TomTom) [reply README.md:40]: Generally I am a bit against the latest logic, but in this case I think is fine to do it. 
The version of the bundler from `Gemfile.lock` is the one used to generate the lockfile.
While we can handle this ok via setup-ruby action in the CI flow, in the documentation installing latest can generate at least some mismatch warnings.

File: .github/workflows/build.yaml [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+jobs:
+  build:
+    name: Buid iOS app
+    runs-on: macos-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      # Ensure we have repos dir created 
+      # Workaround for https://github.com/jfrog/cocoapods-art/issues/32
+      - run: mkdir -p ~/.cocoapods/repos
```
[the remark this answers is not in the document]
Comment (AlexandruLazarescu-TomTom) [reply .github/workflows/build.yaml]: done in 9fbfabb2aa18cbfb3410628c02bdf50da389ad79

File: .github/workflows/build.yaml [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+
+      - name: Setup Ruby
+        uses: ruby/setup-ruby@v1
```
[the remark this answers is not in the document]
Comment (AlexandruLazarescu-TomTom) [reply .github/workflows/build.yaml:20]: done in 9fbfabb2aa18cbfb3410628c02bdf50da389ad79

File: Gemfile.lock [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+
+DEPENDENCIES
+  cocoapods (= 1.11.3)
+  cocoapods-art
+
+RUBY VERSION
+   ruby 2.7.6p219
+
+BUNDLED WITH
+   2.3.23
```
[the remark this answers is not in the document]
Comment (AlexandruLazarescu-TomTom) [reply Gemfile.lock]: see https://github.com/tomtom-international/tomtom-navigation-ios-examples/pull/2#discussion_r1182532193
this is tracking the bundle used during install. Has the benefit of instructing `setup-ruby` which version to install.

File: Gemfile [hunk 9]
```diff
@@ -0,0 +1,4 @@
+source 'https://rubygems.org'
+ruby "~>#{File.read(File.join(__dir__, ".ruby-version")).strip}"
+gem "cocoapods", "1.11.3"
```
[the remark this answers is not in the document]
Comment (AlexandruLazarescu-TomTom) [reply Gemfile:3]: it's not the latest indeed. haven't tried to use the latest. afaik this is the version used internally, so I would keep this one.

**Pull request merged**